### ISSUE TITLE
Orphaned ItemView tab on reload

### ISSUE THREAD

**Issue by MrBertie** [pjeby/hot-reload#22]
I am developing a plugin that creates an ItemView in the sidebar.  When I use hot-reload to disable/enable my plugin it always leaves an orphaned ItemView tab behind and creates a 2nd new one instead.  If I manually disable and re-enable the plugin in the settings this does not happen.  
Any ideas what could be causing this and how I might trouble-shoot this?

**Comment by pjeby** [comment]
If you have a link to the source I can take a look, but I don't know of anything that would do this off the top of my head.  Are you using the `ensureSideLeaf()` API?

**Comment by pjeby** [comment]
Hm.  You have some code to remove the old leaf, but that shouldn't be necessary; recent versions of Obsidian put up a temporary view proxy that it'll refresh with the real view once 1) the plugin is loaded, and 2) the leaf is revealed.

However, it's possible that there's been a change in *how* that gets done, in such a way that Hot Reload isn't triggering the mechanism any more.  (Though that seems unlikely.)  What version of Obsidian are you on, precisely?

The only other thing I can think of is that using an async onload method can result in some weirdness if you save a change at the same time a previous hot reload is occurring, but it's unlikely you'd trigger such a thing *consistently*, I don't think.

**Comment by MrBertie** [comment]
I agree it should not be necessary, but right now this is just a workaround!

I was also wondering if it has anything to do with the new 'Deferred Views' feature.  I am on the latest version (1.7.7 update and installer).

It is consistent. If I manually disable / enable the plugin after a code change it refreshes fine.  If I use *hot-reload* I always get a 2nd orphaned View tab.

**Comment by MrBertie** [comment]
I made the following changes and now it works correctly... Not really sure I can explain why 🤔

```js
    async reload(plugin) {
        const plugins = this.app.plugins;  // <-- added this. as using app is deprecated.

        // Don't reload disabled plugins
        if (!plugins.enabledPlugins.has(plugin)) return;

        await plugins.disablePluginAndSave(plugin);  // <-- changed to the ...AndSave() version
        console.debug("disabled", plugin);
```

**Comment by pjeby** [comment]
Is this still an issue with the current version?
